app/controllers/application_controller.rb

```diff
@@ -2,4 +2,50 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user, :current_order, :current_user_is_owner, :require_login
+
+  def current_user
+      @user ||= User.find_by(id: session[:user_id])
+  end
+
+  def current_user_is_owner
+    session[:user_id] == params[:user_id]
+  end
+
+  def current_order
```

Comment:
Great job laying out the logic of this method and taking it piece by piece!
In production code you would take out those comments since your code is so beautifully readable. 
But I like them in educational code because it helps me see your thought process.

Extension exercise:
Consider if there is a way to set up the control flow (if-else statements) to make it so that you don't have to update the order. 

Database calls are some of the slowest things in an application so writing as few times as possible is a good thing to consider.